swift/ql/lib/codeql/swift/generated/Comment.qll

```diff
@@ -1,8 +1,10 @@
 // generated by codegen/codegen.py
+private import codeql.swift.generated.Synth
+private import codeql.swift.generated.Raw
 import codeql.swift.elements.Locatable
 
-class CommentBase extends @comment, Locatable {
+class CommentBase extends Synth::TComment, Locatable {
   override string getAPrimaryQlClass() { result = "Comment" }
 
-  string getText() { comments(this, result) }
+  string getText() { result = Synth::toRawComment(this).(Raw::Comment).getText() }
```

Comment:
Why does `Synth::toRawComment` not specify a `Raw::Comment` return type?

Reply:
I'll try it out though, maybe it can work

Reply:
Yeah, fair enough. Ruby just has a single predicate called `to/fromGenerated` (in which case the generic result type kinda makes sense). Should we do the same?

Reply:
I think it could work, but it would require a bit of an effort and even more complexity on the generation side, so I'd rather not do it unless we have reason to. As far as I can tell we won't be calling these conversion functions a lot in non-generated code.

Reply:
(I will rename the conversion functions thought to avoid confusion)

Reply:
The "I think it could work" comment above was written without seeing your last comment and was referring to specialising the raw types on conversions.

As to


> Yeah, fair enough. Ruby just has a single predicate called `to/fromGenerated` (in which case the generic result type kinda makes sense). Should we do the same (just define a giant disjunction in a single predicate, that is)?

I had that version at one point, it was working ok. I then tried the version with separate conversion functions and it turned out it was performing a bit better. Not by a lot, but still noticeably, probably exactly because avoiding the giant disjunction in many cases.